### PR TITLE
feat(material-experimental/mdc-chips): add harness skeleton

### PR DESCRIPTION
**Pull request by andrewseguin**
Covers the skeleton work for harnesses for the grid + rows + input, listbox + options, and set + chips.

### CODE REVIEW

File: src/material-experimental/mdc-chips/harness/chip-grid-harness.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  private _rows = this.locatorForAll(MatChipRowHarness);
+  private _input = this.locatorFor(MatChipInputHarness);
+
+  /** Gets promise of the harnesses for the chip rows. */
+  async getRows(): Promise<MatChipRowHarness[]> {
+    return await this._rows();
+  }
+
+  /** Gets promise of the chip input harness. */
+  async getInput(): Promise<MatChipInputHarness|null> {
```
Comment (jelbourn) [inline src/material-experimental/mdc-chips/harness/chip-grid-harness.ts]: What do you think of `getTextInput`? I'm thinking that just "input" by itself might be too ambiguous

File: src/material-experimental/mdc-chips/harness/chip-harness.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * Harness for interacting with a mat-chip in tests.
+ * @dynamic
+ */
+export class MatChipHarness extends ComponentHarness {
+  static hostSelector = 'mat-basic-chip, mat-chip';
+
+  /** Gets a promise for the text content the option. */
+  async getTextContent(): Promise<string> {
```
Comment (jelbourn) [inline src/material-experimental/mdc-chips/harness/chip-harness.ts]: Just `getText`? That's what I used for button and I think we should probably be consistent

File: src/material-experimental/mdc-chips/harness/chip-harness.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    return (await this.host()).text();
+  }
+
+  /** Gets a promise for the disabled state. */
+  async isDisabled(): Promise<boolean> {
+    const ariaSelected = (await this.host()).getAttribute('aria-disabled');
```
Comment (jelbourn) [inline src/material-experimental/mdc-chips/harness/chip-harness.ts]: `ariaSelected`?

File: src/material-experimental/mdc-chips/harness/chip-harness.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+export class MatChipHarness extends ComponentHarness {
+  static hostSelector = 'mat-basic-chip, mat-chip';
+
+  /** Gets a promise for the text content the option. */
+  async getTextContent(): Promise<string> {
+    return (await this.host()).text();
+  }
+
+  /** Gets a promise for the disabled state. */
+  async isDisabled(): Promise<boolean> {
```
Comment (jelbourn) [inline src/material-experimental/mdc-chips/harness/chip-harness.ts]: I'm not sure about this one. A chip by itself doesn't have any interactive role, and thus being disabled doesn't make sense.

File: src/material-experimental/mdc-chips/harness/chip-input.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+import {ComponentHarness} from '@angular/cdk-experimental/testing';
+
+/**
+ * Harness for interacting with a grid's chip input in tests.
+ * @dynamic
+ */
+export class MatChipInputHarness extends ComponentHarness {
+  static hostSelector = 'input.mat-mdc-chip-input';
```
Comment (jelbourn) [inline src/material-experimental/mdc-chips/harness/chip-input.ts]: Is the `input` part necessary?

File: src/material-experimental/mdc-chips/harness/chip-input.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+  /** Gets a promise for the disabled state. */
+  async isDisabled(): Promise<boolean> {
+    const ariaSelected = (await this.host()).getAttribute('disabled');
+    return await ariaSelected === 'true';
```
Comment (jelbourn) [inline src/material-experimental/mdc-chips/harness/chip-input.ts]: Just one line?
```ts
return await ((await this.host()).getAttribute('disabled')) === 'true';
```

File: src/material-experimental/mdc-chips/harness/chip-listbox-harness.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  private _options = this.locatorForAll(MatChipOptionHarness);
+
+  /** Gets promise of the harnesses for the chip options in the listbox. */
+  async getOptions(): Promise<MatChipOptionHarness[]> {
+    return await this._options();
+  }
+
+  /** Gets promise of the selected options. */
+  async getSelected(): Promise<MatChipOptionHarness[]> {
+    const options = await this._options();
```
Comment (jelbourn) [inline src/material-experimental/mdc-chips/harness/chip-listbox-harness.ts]: Simplify?
```ts
return (await this._options()).filter(o => await o.isSelected());
```
(`async` functions automatically wrap non-promise return values into Promises)

File: src/material-experimental/mdc-chips/harness/chip-grid-harness.ts (hunk 1 above)
Comment (andrewseguin) [reply src/material-experimental/mdc-chips/harness/chip-grid-harness.ts]: SGTM

File: src/material-experimental/mdc-chips/harness/chip-input.ts (hunk 5 above)
Comment (andrewseguin) [reply src/material-experimental/mdc-chips/harness/chip-input.ts]: No, can't remember why I included that but it doesn't seem necessary

File: src/material-experimental/mdc-chips/harness/chip-input.ts (hunk 6 above)
Comment (andrewseguin) [reply src/material-experimental/mdc-chips/harness/chip-input.ts]: It's not as readable to me but perhaps my brain is still adjusting to awaits

File: src/material-experimental/mdc-chips/harness/chip-harness.ts (hunk 4 above)
Comment (andrewseguin) [reply src/material-experimental/mdc-chips/harness/chip-harness.ts]: Maybe we should move this to the `MatChipOptionHarness`? Should we not support `disabled` on the base `MatChip`?

Comment (andrewseguin) [reply src/material-experimental/mdc-chips/harness/chip-harness.ts]: I moved it to the option, we can evaluate whether the MdcMatChip should have `disabled` as an input

**Pull request merged**